modules/basics/src/main/java/com/opengamma/strata/basics/market/MarketData.java

```diff
@@ -5,71 +5,69 @@
  */
 package com.opengamma.strata.basics.market;
 
-import java.util.Map;
+import java.time.LocalDate;
 
 import com.opengamma.strata.collect.timeseries.LocalDateDoubleTimeSeries;
 
 /**
- * A set of market data used when calculating values for a single scenario.
+ * Provides access to market data, such as curves, surfaces and time-series.
+ * <p>
+ * Market data is looked up using subclasses of {@link MarketDataKey}.
+ * All data is valid for a single date, defined by {@link #getValuationDate()}.
+ * When performing calculations with scenarios, only the data of a single scenario is accessible.
+ * <p>
+ * See {@link ImmutableMarketData} for a standalone implementation.
  */
 public interface MarketData {
 
   /**
-   * Checks if this set of data contains a value for the specified key.
-   *
-   * @param key  a key identifying an item of market data
-   * @return true if this set of data contains a value for the specified key
+   * Gets the valuation date of the market data.
+   * <p>
+   * All values accessible through this interface have the same valuation date.
+   * 
+   * @return the valuation date
    */
-  public abstract boolean containsValue(MarketDataKey<?> key);
+  public abstract LocalDate getValuationDate();
 
+  //-------------------------------------------------------------------------
   /**
-   * Checks if this set of data contains a time series for the specified key.
+   * Checks if this set of data contains a value for the specified key.
    *
-   * @param key  a key identifying an item of market data
-   * @return true if this set of data contains a time series of market data for the specified key
+   * @param <T>  the type of the market data
+   * @param key  the key identifying the item of market data
+   * @return true if this set of data contains a value for the specified key
    */
-  public abstract boolean containsTimeSeries(ObservableKey key);
+  public abstract <T> boolean containsValue(MarketDataKey<T> key);
 
   /**
-   * Returns a market data value.
+   * Gets the market data value identified by the specified key.
    * <p>
-   * The date of the market data is the same as the valuation date of the calculations.
+   * The result will be a single piece of market data valid for the valuation date.
    *
-   * @param <T>  type of the market data
-   * @param key  key identifying the market data
-   * @return a market data value
-   * @throws IllegalArgumentException if there is no value for the specified key
+   * @param <T>  the type of the market data
+   * @param key  the key identifying the item of market data
+   * @return the market data value
+   * @throws IllegalArgumentException if no value is found
+   * @throws RuntimeException if an unexpected error occurs
```

Comment:
This comment seems a bit redundant. You could apply it to any method in any piece of Java and it would be correct.